tests/test_live.py

```diff
@@ -0,0 +1,340 @@
+
+    assert rod.valid
+    assert segment_equal(s, rod.db)
+
+
+def test_create_group(ws, canvas, cleanup):
+    r = Rect[0:0.1, 0.2:0.3, Layer('M1', 'drawing')]
+    s = Segment.from_start_end(Point(1, 1), Point(2, 1), 0.1, Layer('M2', 'pin'))
+    g = Group([r, s])
+
+    canvas.append(g)
+    (db,) = canvas.draw()
+
+    assert db.valid
+    assert rect_equal(g.bbox, db.db)
+    assert rect_equal(db.db.figs[0], r)
+    assert segment_equal(db.db.figs[1], s)
+
+
+def test_create_nested_group(ws, canvas, cleanup):
+    one = Rect[0:0.1, 0.2:0.3, Layer('M1', 'drawing')]
+    two = one.copy()
+    two.translate(left=one.right)
+    group_one = Group([one, two])
+
+    three = one.copy()
+    three.translate(top=one.bottom)
+
+    group_two = Group([group_one, three])
+
+    canvas.append(group_two)
+    (db,) = canvas.draw()
+
+    assert rect_equal(group_two.bbox, db.db)
+    assert rect_equal(db.db.figs[1], three)
+    assert rect_equal(db.db.figs[0].figs[0], one)
+    assert rect_equal(db.db.figs[0].figs[1], two)
+
+
+def test_delete_works(ws, canvas, cleanup):
+    one = Rect[0:0.1, 0.2:0.3, Layer('M1', 'drawing')]
+    group = Group([one])
+
+    canvas.append(group)
+    (db,) = canvas.draw()
+
+    rect_db = DbShape(db.db.figs[0])
+
+    rect_db.delete()
+
+    assert not db.db.figs
+    assert not rect_db.valid
+    assert db.valid
+
+
+def test_align(ws, cv, cell_cv):
+    """
+    Crosses all alignable objects inheriting from
+    figure with PR alignment for instances, align handles and maintain flag for the align method.
+    """
+    dummy_layer = Layer("M1", "drawing")
+
+    @dataclass
+    class Alignable:
+        shape: Figure
+        align_at: str
+        rect: Rect
+
+        @property
+        def align(self):
+            if self.align_at == "b_box":
+                return self.shape.b_box
+            elif self.align_at == "pr_boundary":
+                return self.shape.pr_boundary
+
+        @property
+        def b_box(self) -> BoundingBox:
+            if self.align_at == "b_box":
+                return self.shape.skill_b_box
+            elif self.align_at == "pr_boundary":
+                return self.shape.skill_pr_boundary
+
+    count = 0
+
+    @contextmanager
+    def create_inst():
+        nonlocal count, ws, cv, cell_cv
+        name = f"I{count}_test"
+        db = ws.db.create_inst(cv, cell_cv, name, (0, 0), "R0")
+        count += 1
+        try:
+            shape = Instance.from_name(cv, name)
+            yield Alignable(shape=shape, align_at="b_box", rect=Rect[-1:11, -3:7])
+        finally:
+            ws.db.delete_object(db)
+
+    @contextmanager
+    def create_pr_inst():
+        # Instance to be aligned at the P&R boundary
+        with create_inst() as inst:
+            yield Alignable(shape=inst.shape, align_at="pr_boundary", rect=Rect[-1:7, -3:5])
+
+    @contextmanager
+    def create_inst_group():
+        with create_inst() as i1:
+            with create_inst() as i2:
```

Comment:
`with create_inst() as i1, create_inst() as i2:`
Flat is better.

Why `i1` `i2` is there no better name?